### PR TITLE
[WIP] fix scroll area about scrollHeight and scrollWidth

### LINKED ISSUE #19280: element.clientHeight & element.scrollHeight  incomplete implement
With the following `test.html`:
```
    <body>
        <div id="scroller" style="height: 100px; width: 100px; overflow: scroll; background: red;">
            <div style="background: green; margin-top: 100px; width: 100px; height: 100px;"></div>
        </div>
        <div id='show'></div>
        <script>
            var scroller = document.getElementById("scroller");
            console.log(scroller.clientHeight)
            console.log(scroller.scrollHeight) 
        </script>
    </body>
```
Firefox will get:
```
96
200
```

Servo will get:
```
100
100
```

This might be an issue related to #19268

### PR DESCRIPTION
**Pull request by tigercosmos**
Currently `scrollHeight` will get the result that `max(height + padding, margin)`
It should be `scrollHeight = height + padding + margin`
Same as `scrollWidth`

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #19280 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19469)

### CODE REVIEW

**Comment by tigercosmos** [comment]
@KiChjang  This make many `TIMEOUT` > http://build.servo.org/builders/linux-rel-wpt/builds/7147
Do you have any idea why causes that?

**Comment by jdm** [comment]
That is a problem affecting many PRs right now that we don't understand yet :(

**Comment by tigercosmos** [comment]
@jdm This PR has many `PASS`, but is also has many `TIMEOUT`. How could I determine this change is fine? Is `TIMEOUT` intermittent? maybe try again?

File: components/layout/query.rs [hunk 1]
```diff
@@ -463,16 +463,16 @@ impl FragmentBorderBoxIterator for UnioningFragmentScrollAreaIterator {
         let bottom_padding = (border_box.size.height.to_f32_px() - bottom_border - top_border) as i32;
```
Comment (emilio) [inline components/layout/query.rs]: This looks somewhat fishy to me, in the sense that `margin` accounts for writing-mode, but not `padding`... Not that it wasn't wrong before, though...

File: components/layout/query.rs [hunk 2]
```diff
@@ -639,18 +639,18 @@ pub fn process_node_scroll_area_request< N: LayoutNode>(requested_node: N, layou
             let bottom = max(iterator.union_rect.size.height, iterator.origin_rect.size.height);
             let left = max(iterator.union_rect.origin.x, iterator.origin_rect.origin.x);
             Rect::new(Point2D::new(left, iterator.origin_rect.origin.y),
-                      Size2D::new(iterator.origin_rect.size.width, bottom))
+                      Size2D::new(iterator.union_rect.size.width, bottom))
```
Comment (emilio) [inline components/layout/query.rs]: So, this also includes left and top margins, right? That looks wrong per https://drafts.csswg.org/cssom-view/#scrolling-area, isn't it?

**Comment by emilio** [comment]
All the passes and timeouts are known-intermittent.

**Comment by tigercosmos** [comment]
@emilio this commit is for `scroll area`, and I try before.
#19478 need to fix first

File: components/layout/query.rs (hunk 1 above)
Comment (tigercosmos) [reply components/layout/query.rs]: Are you saying `bottom_padding` here is weird? What is `writing-mode`?

**Comment by tigercosmos** [comment]
```json
{"status": "FAIL", "group": "default", "message": "assert_equals: scrollWidth should be 700 expected 700 but got 660", "stack": "nonQuirksFrame.onload/<@http://web-platform.test:8000/css/cssom-view/scrolling-quirks-vs-nonquirks.html:189:5\nTest.prototype.step@http://web-platform.test:8000/resources/testharness.js:1489:20\ntest@http://web-platform.test:8000/resources/testharness.js:511:9\nnonQuirksFrame.onload@http://web-platform.test:8000/css/cssom-view/scrolling-quirks-vs-nonquirks.html:188:3\n", "subtest": "scrollWidth/scrollHeight on the HTML body element in non-quirks mode", "test": "/css/cssom-view/scrolling-quirks-vs-nonquirks.html", "line": 128714, "action": "test_result", "expected": "PASS"}
{"status": "FAIL", "group": "default", "message": "assert_equals: initial scrollWidth should be 510 expected 510 but got 310", "stack": "@http://web-platform.test:8000/css/cssom-view/elementScroll-002.html:70:9\nTest.prototype.step@http://web-platform.test:8000/resources/testharness.js:1489:20\ntest@http://web-platform.test:8000/resources/testharness.js:511:9\n@http://web-platform.test:8000/css/cssom-view/elementScroll-002.html:67:5\n", "subtest": "Element scrollHeight/scrollWidth getter test with padding + height/width + margin", "test": "/css/cssom-view/elementScroll-002.html", "line": 148747, "action": "test_result", "expected": "PASS"}
{"status": "FAIL", "group": "default", "message": "assert_equals: initial scrollWidth should be 500 expected 500 but got 300", "stack": "@http://web-platform.test:8000/css/cssom-view/elementScroll-002.html:76:9\nTest.prototype.step@http://web-platform.test:8000/resources/testharness.js:1489:20\ntest@http://web-platform.test:8000/resources/testharness.js:511:9\n@http://web-platform.test:8000/css/cssom-view/elementScroll-002.html:73:5\n", "subtest": "Element scrollHeight/scrollWidth getter test with height/width + margin", "test": "/css/cssom-view/elementScroll-002.html", "line": 148748, "action": "test_result", "expected": "PASS"}
```

**Pull request closed without merging**